### DISCUSSION TITLE
Credentials storage

### DISCUSSION THREAD

**Discussion by mattjbr123** [NERC-CEH/rse_group#20] [General]
Thinking around how best to store credentials, such as those for object storage, in public GitHub repos or other situations where access to the credentials needs to be managed

**Comment by mattjbr123** [comment]
Datalabs developers also interested in finding good solutions to this: https://nerc-datalabs.slack.com/archives/C85TV1MAR/p1715610602635919

**Comment by mattjbr123** [comment]
https://blog.gitguardian.com/how-to-handle-secrets-in-jupyter-notebooks/ has some suggestions
I'm particularly interested in the AWS Secrets Manager section, which I'll take a look at sometime...

**Comment by metazool** [comment]
[nice overview blog post about sliding scale of notebook secret safety](https://blog.gitguardian.com/how-to-handle-secrets-in-jupyter-notebooks/) but it ends up with proprietary cloud based expressions of managing a keyvault

When in doubt about candidates for open source component choice I check what the [NCSA DecentCI](https://decentci.ncsa.illinois.edu/) research infrastructure project has picked out - they don't have one in this case! They list [bitnami's sealed-secrets](https://github.com/bitnami-labs/sealed-secrets) for Kubernetes (and our on-prem instance is running it already), and list other Hashicorp components so [Hashicorp vault](https://github.com/hashicorp/vault) could be a candidate

How does one make the case for getting things like this into the DataLabs backlog? It's still very much a "flagship" offering for Digital Research Infrastructure projects....
